scripts/main.js

```diff
@@ -0,0 +1,326 @@
+    var size = gui.add(fizzyText, "pointWidth", 0.1, 2).name("Point width");
+    var mode_chooser = gui.add(fizzyText, "mode", ["offline", "online"]).name("Mode");
+    gui.add(fizzyText, "reset_defaults").name("Reset defaults");
+
+    var playback = gui.addFolder("Playback");
+    var pcontrol = playback.add(fizzyText, "play").name("Play").listen();
+    playback.add(fizzyText, "restart").name("Restart");
+    var chooser = playback.add(fizzyText, "currentSave", allSaves).name("Choose save");
+
+    playback.open();
+
+
+    mode_chooser.onChange(function(value) {
+        if (value === "offline") { data = fileGet("lib/saves/" + fizzyText.currentSave + ".json"); }
+        FrameId = 2;
+        fizzyText.play = true;
+        while (scene.children.length > 5) {
+            var selectedObject = scene.getObjectByName("point");
+            scene.remove(selectedObject);
+        }
+        arrayOfPoints = [];
+
+    });
+
+    chooser.onChange(function(value) {
+        FrameId = 2;
+        fizzyText.play = true;
+        while (scene.children.length > 5) {
+            var selectedObject = scene.getObjectByName("point");
+            scene.remove(selectedObject);
+        }
+        arrayOfPoints = [];
+        data = fileGet("lib/saves/" + fizzyText.currentSave + ".json");
+    });
+
+    size.onChange(function(value) {
+        refreshArrayOfPoints(fizzyText, arrayOfPoints);
+    });
+
+    cl.onChange(function(value) {
+        fizzyText.color[0] = value[0];
+        fizzyText.color[1] = value[1];
+        fizzyText.color[2] = value[2];
+        refreshArrayOfPoints(fizzyText, arrayOfPoints);
+    });
+
+
+
+    var tmp = init();
+    var scene = tmp[0];
+    var renderer = tmp[1];
+    var camera = tmp[2];
+    var controls = tmp[3];
+
+    // run game loop (update, render, repeat)
+    data = fileGet("/lib/saves/" + fizzyText.currentSave + ".json");
+    var GameLoop = function() {
+        requestAnimationFrame(GameLoop);
+        stats.begin();
+        controls.update();
+
+        if (fizzyText.mode === "offline") {
+            var trigger = {"status": true};
+        } else {
+            var trigger = httpGet("/get_status");
+            data = httpGet("/get_frame");
+        }
+
+        if (FrameId === 2 && fizzyText.mode === "offline") {
+            speed = 1;
+            fizzyText.play = true;
+            arrayOfPoints = [];
+            while (scene.children.length > 5) {
+                var selectedObject = scene.getObjectByName("point");
+                scene.remove(selectedObject);
+            }
+        }
+
+        if (FrameId === data.length - 1) {
+            speed = -1;
+        }
+
+        if (fizzyText.play && FrameId !== 0 && fizzyText.mode === "offline") {
+            FrameId += speed;
+        }
+
+        if (trigger["status"] && data.length - 1 !== FrameId) {
+            if (fizzyText.mode === "offline") {
+                fizzyText.particles = data[FrameId]["x"].length;
```

Comment:
Something is wrong with either logic or variables naming: `particles` should be an array or something alike, but is assigned to an integer